ztrouy/cpyou
# Feature - Improved Build and Components System for Backend

### Purpose
To have each type of computer component have their own model, enabling future features such as checking component compatability, PSU sufficiency, and port/header sufficiency.

### Files Changed
- Removed all files related to generic Components and Build Components, including `Component.cs`, `ComponentDTO.cs`, `ComponentForBuildDTO.cs`, ComponentNoNavDTO.cs`, `BuildComponent.cs`, `BuildComponentDTO.cs`, BuildComponentCreateDTO.cs`, `BuildComponentForEditFormDTO.cs`, and `ComponentsController.cs`
- Created initial model and DTOs for Interfaces in new `Interface.cs`, `InterfaceDTO.cs`, and `InterfaceNoNavDTO.cs`
- Created initial model, DTOs, and Controller for the new CPU component in new `CPU.cs`, `CPUDTO.cs`, `CPUForBuildDTO.cs`, `CPUNoNavDTO.cs`, and `CPUController.cs`
- Created initial model, DTOs, and Controller for the new GPU component in new `GPU.cs`, `GPUDTO.cs`, `GPUForBuildDTO.cs`, `GPUNoNavDTO.cs`, and `GPUController.cs`
- Created initial model, DTOs, and Controller for the new Power Supply component in new `PSU.cs`, `PSUDTO.cs`, `PSUNoNavDTO.cs`, and `PSUController.cs`
- Created initial model, DTOs, and Controller for the new CPU Cooler component in new `Cooler.cs`, `CoolerDTO.cs`, `CoolerNoNavDTO.cs`, and `CoolerController.cs`
- Created initial model, DTOs, and Controller for the new Motherboard component in new `Motherboard.cs`, `MotherboardDTO.cs`, `MotherboardForBuildDTO.cs`, `MotherboardNoNavDTO.cs`, and `MotherboardController.cs`
- Created initial model, DTOs, and Controller for the new Memory module component in new `Memory.cs`, `MemoryDTO.cs`, `MemoryForBuildDetailsDTO.cs`, `MemoryForBuildMemoryDTO`, `MemoryNoNavDTO.cs`, and `MemoryController.cs`
- Created initial model and DTOs for the BuildMemory bridge table in new `BuildMemory.cs`, `BuildMemoryDTO.cs`, `BuildMemoryCreateDTO.cs`, `BuildMemoryForBuildDTO.cs`, and `BuildMemoryForEditFormDTO.cs`
- Created initial model, DTOs, and Controller for the new Storage device component in new `Storage.cs`, `StorageDTO.cs`, `StorageForBuildDetailsDTO.cs`, `StorageForBuildStorageDTO`, `StorageNoNavDTO.cs`, and `StorageController.cs`
- Created initial model and DTOs for the BuildStorage bridge table in new `BuildStorage.cs`, `BuildStorageDTO.cs`, `BuildStorageCreateDTO.cs`, `BuildStorageForBuildDTO.cs`, and `BuildStorageForEditFormDTO.cs`
- Updated Build model, DTOs, and Controller to use new build system with bespoke models for each Component type in `Build.cs`, `BuildDTO.cs`, `BuildCreateDTO.cs`, `BuildEditDTO.cs`, `BuildForBuildDetailsDTO.cs`, `BuildForEditFormDTO.cs`, `BuildForListDTO.cs`, and `BuildController.cs`
- Replaced DbSets and seed data of general Components with those for new bespoke Components, and replaced seed data for Builds, Comments, and Replies in `CPYouDbContext.cs`

### To Test
Fetch, setup, and run just the backend according to project README, then confirm the following in Swagger/Postman
- Able to fetch all CPUs, GPUs, PSUs, Coolers, Motherboards, Memory modules, and Storage devices
- Able to successfully use all Build endpoints

closes #29
